edisgo/grid/components.py

```diff
@@ -448,45 +723,125 @@ def timeseries(self):
             # get time series for active power depending on if they are
             # differentiated by weather cell ID or not
             if isinstance(self.grid.network.timeseries.generation_fluctuating.
-                          columns, pd.MultiIndex):
+                                  columns, pd.MultiIndex):
                 if self.weather_cell_id:
                     try:
-                        timeseries = self.grid.network.timeseries.\
+                        timeseries = self.grid.network.timeseries. \
                             generation_fluctuating[
                             self.type, self.weather_cell_id].to_frame('p')
                     except KeyError:
                         logger.exception("No time series for type {} and "
                                          "weather cell ID {} given.".format(
-                                            self.type, self.weather_cell_id))
+                            self.type, self.weather_cell_id))
                         raise
                 else:
                     logger.exception("No weather cell ID provided for "
                                      "fluctuating generator {}.".format(
-                                        repr(self)))
+                        repr(self)))
                     raise KeyError
             else:
                 try:
-                    timeseries = self.grid.network.timeseries.\
+                    timeseries = self.grid.network.timeseries. \
                         generation_fluctuating[self.type].to_frame('p')
                 except KeyError:
                     logger.exception("No time series for type {} "
                                      "given.".format(self.type))
                     raise
 
-            timeseries['q'] = timeseries['p'] * tan(acos(self.power_factor))
             timeseries = timeseries * self.nominal_capacity
 
```

Comment:
Make sure the subtract curtailment at this point, so that the correct reactive power is calculated from the `power_factor` also considers the reduced active power.